skxray/diff_roi_choice.py

```diff
@@ -0,0 +1,364 @@
+
+    delta_r : float
+        thickness of the ring
+
+    *step_r : tuple
+        step value for the next ring from the end of the previous
+        ring.
+        same step - same step values between rings (one value)
+        different steps - different step value between rings (provide
+        step value for each ring eg: 6 rings provide 5 step values)
+
+    Returns
+    -------
+    ring_vals : ndarray
+        edge values of the required rings
+
+    ring_inds : ndarray
+        indices of the required rings
+        after discarding zero values from the shape
+        ([detector_size[0]*detector_size[1]], )
+
+    num_pixels : ndarray
+        number of pixels in each ring
+
+    pixel_list : ndarray
+        pixel indices for the required rings
+    """
+
+    grid_values = _grid_values(img_dim, calibrated_center)
+
+    ring_vals = []
+
+    for arg in step_r:
+        if arg < 0:
+            raise ValueError("step value for the next ring from the "
+                             "end of the previous ring has to be positive ")
+
+    if len(step_r) == 1:
+        #  when there is a same values of step between rings
+        #  the edge values of rings will be
+        ring_vals = first_r + np.r_[0, np.cumsum(np.tile([delta_r,
+                                                          float(step_r[0])],
+                                                         num_rings))][:-1]
+    elif len(step_r) == (num_rings - 1):
+        #  when there is a different step values between each ring
+        #  edge values of the rings will be
+        ring_vals.append(first_r)
+        for arg in step_r:
+            ring_vals.append(ring_vals[-1] + delta_r)
+            ring_vals.append(ring_vals[-1] + float(arg))
+        ring_vals.append(ring_vals[-1] + delta_r)
+
+    else:
+        raise ValueError("Provide valid step value between rings"
+                         "For different step values between rings-"
+                         "provide step value for each q ring ")
+
+    # indices of rings
+    ring_inds = np.digitize(np.ravel(grid_values), np.array(ring_vals))
+
+    # to discard every-other bin and set the discarded bins indices to 0
+    ring_inds[ring_inds % 2 == 0] = 0
+    # change the indices of odd number of rings
+    indx = ring_inds > 0
+    ring_inds[indx] = (ring_inds[indx] + 1) // 2
+
+    (ring_inds, ring_vals, num_pixels,
+     pixel_list) = _process_rings(num_rings, img_dim,
+                                  ring_vals, ring_inds)
+
+    return ring_inds, ring_vals, num_pixels, pixel_list
+
+
+def _grid_values(img_dim, calibrated_center):
+    """
+    Parameters
+    ----------
+    img_dim: tuple
+        shape of the image (detector X and Y direction)
+        shape is [detector_size[0], detector_size[1]])
+
+    calibarted_center : tuple
+        defining the (x y) center of the image (mm)
+    """
+
+    xx, yy = np.mgrid[:img_dim[0], :img_dim[1]]
+    x_ = (xx - calibrated_center[0])
```

Comment:
@tacaswell, The bug with the plots (There is an off-by-one bug in here:)
, it was due to earlier version of it ,  ( Earlier x_ and y_  was as following)
x_ = (xx - calibrated_center[0]+1
y_ = (yy - calibrated_center[1] + 1)